skyportal/handlers/api/source.py

```diff
@@ -882,9 +885,24 @@ class Validator(Schema):
             obj_query = obj_query.filter(Obj.within(other, radius))
         if start_date:
             start_date = arrow.get(start_date.strip()).datetime
-            obj_query = obj_query.filter(
-                Obj.last_detected_at(self.current_user) >= start_date
-            )
+            if query_mode == 1:
+                obj_query = obj_query.filter(
+                    Obj.last_detected_at(self.current_user) >= start_date
+                )
+            elif query_mode == 2:
+                utc_timestamp = start_date.timestamp()
+                mjd = utc_timestamp / 86400 + 40587  # from 1970 Jan 1
```

Comment:
We use `astropy.time` for mjd calculations (helps avoid off by 0.5 day errors). For example:
```
mjd = Time(request.payload["start_date"], format='iso').mjd
```